packages/cli/src/modules/insights/insights-pruning.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Service } from '@n8n/di';
+import { Logger } from 'n8n-core';
+
+import { InsightsByPeriodRepository } from './database/repositories/insights-by-period.repository';
+import { InsightsConfig } from './insights.config';
+
+@Service()
+export class InsightsPruningService {
+	private pruneInsightsTimer: NodeJS.Timer | undefined;
+
+	constructor(
+		private readonly insightsByPeriodRepository: InsightsByPeriodRepository,
+		private readonly config: InsightsConfig,
+		private readonly logger: Logger, // Assuming a logger is injected,
+	) {
+		this.logger = this.logger.scoped('insights');
+	}
+
+	get isPruningEnabled() {
+		return this.config.maxAgeDays > -1;
+	}
+
+	startPruningTimer() {
+		if (!this.isPruningEnabled) {
+			return;
+		}
```

Comment:
Would it be cleaner to check this at call site?

Reply:
The call was to prevent an unnecessary timer running, but what about checking on call on top of this ?

Reply:
If we want to protect ourselves from calling this more than once, we could have an assertion with `strict` on the absence of the timer, so we enforce that we don't call this more than once, a bit better than allowing it to be called many times and returning early.

Reply:
I'm not sure I follow :thinking: . This check is not to prevent multiple calls, it's just to check if the timer is enabled or not in the configuration.